aeron-driver/src/main/java/uk/co/real_logic/aeron/driver/DriverConductor.java

```diff
@@ -683,6 +684,11 @@ private void onRemoveSubscription(final long registrationId, final long correlat
         final int refCount = channelEndpoint.decRefToStream(subscription.streamId());
         if (0 == refCount)
         {
+            connections
```

Comment:
It wasn't clear which thread should own the ability to write state to the NetworkConnection objects (currently multiple threads appear to).

This code could be removed from the Receiver class and just moved here to the DriverConductor but I'm not sure if other calls expect it to also happen in the Receiver code (like in response to network events).  As reported in issue #141, if it isn't done here, it triggers the bug described in the issue because the "work" will be put on a queue for another thread to do later, which opens a window where new "add subscriber" messages from the client will temporarily get attached to this NetworkConnection before it gets marked inactive on the other thread and then they never recover or receive messages.